predict.py

```diff
@@ -0,0 +1,302 @@
+import sys
+import os
+import tempfile
+import warnings
+from pathlib import Path
+import nltk
+import torch
+from torch import nn
+import torchvision.transforms as transforms
+import numpy as np
+import imageio
+from PIL import Image as Image_PIL
+from scipy.stats import truncnorm
+from nltk.corpus import wordnet as wn
+import cma
+import sklearn.metrics
+import cog
+
+sys.path.insert(0, "stylegan2_ada_pytorch")
+from pytorch_pretrained_biggan import convert_to_images, utils
+import inference.utils as inference_utils
+import data_utils.utils as data_utils
+
+NORM_MEAN = torch.Tensor([0.485, 0.456, 0.406]).view(3, 1, 1)
+NORM_STD = torch.Tensor([0.229, 0.224, 0.225]).view(3, 1, 1)
+
+nltk.download("wordnet")
+IND2NAME = {
+    index: wn.of2ss("%08dn" % offset).lemma_names()[0]
+    for offset, index in utils.IMAGENET.items()
+}
+NAME2IND = dict([(value, key) for key, value in IND2NAME.items()])
+
+CLASS_NAMES = sorted(list(IND2NAME.values()))
+
+
+class Predictor(cog.Predictor):
+    def setup(self):
+        torch.manual_seed(np.random.randint(sys.maxsize))
+        warnings.simplefilter("ignore", cma.evolution_strategy.InjectionWarning)
+        self.last_gen_model = None
+        self.last_feature_extractor = None
+        self.model = None
+        self.feature_extractor = None
+        self.noise_size = 128
+        self.batch_size = 4
+        self.size = 256
+
+    @cog.input("image", type=Path, help="Input image Instance")
+    @cog.input("gen_model", type=str, options=["icgan", "cc_icgan"], default="icgan",
+               help='Select type of IC-GAN model. "icgan" is conditioned on the input image; '
+                    '"cc_icgan" is conditioned on both input image and a conditional_class')
+    @cog.input("conditional_class", type=str, default=None, options=CLASS_NAMES,
+               help="Choose conditional class. Only valid for gen_model=cc_icgan")
+    @cog.input("num_samples", type=int, default=1, options=[1, 4, 9, 16],
+               help="number of samples generated")
+    @cog.input("seed", type=int, default=0, help="seed=0 means no seed")
+    def predict(self, image, gen_model="icgan", conditional_class=None, num_samples=1, seed=0):
+        assert isinstance(seed, int), "seed should be an integer"
+        if gen_model == 'cc_icgan':
+            assert conditional_class is not None, 'please set conditional_class for cc_icgan'
+        num_samples_ranked = num_samples
+        experiment_name = (
+            "icgan_biggan_imagenet_res256"
+            if gen_model == "icgan"
+            else "cc_icgan_biggan_imagenet_res256"
+        )
+        num_samples_total = 160
```

Comment:
I would recommend setting `num_samples_total = num_samples*10` for faster computation when the user wants only 1,4 or 9 images.